app/views/automated_tests/_test_runs.html.erb

```diff
@@ -20,11 +20,18 @@
 
 <%# student results page (with instructor tests released) does not need the button %>
 <% unless @current_role.student? && ran_by == 'instructor' %>
-  <% route = @current_role.instructor? ?
+  <%
+    route = @current_role.instructor? ?
     run_tests_course_result_path(@current_course, @result) :
-    execute_test_run_course_assignment_automated_tests_path(@current_course, @assignment) %>
+    execute_test_run_course_assignment_automated_tests_path(@current_course, @assignment)
+
+    test_run_id = @grouping.test_runs.find_by(status: 0)&.autotest_test_id
```

Comment:
This query is on the right track, but needs some changes:

- the query should also only look for test runs that were created by the current role (i.e., current student)
- don't hard-code 0; instead, read about enums at https://guides.rubyonrails.org/active_record_querying.html#enums (since `status` is an enum field)
- you should be retrieving just the `id`, not `autotest_test_id`